app/src/main/index.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+import { app, BrowserWindow }  from 'electron'
+import { format as formatUrl } from 'url'
+import * as path               from 'path'
+
+const is_development = process.env.NODE_ENV !== 'production'
+
+let main_window_keep_alive
+
+function create_main_window() {
+    const window = new BrowserWindow({
+        width  : 1024,
+        height : 768,
+        frame  : false,
+    })
+
+    let debug_scene = "";
+    let command     = "debug";
+    if (app.commandLine.hasSwitch(command)) {
+        let value = app.commandLine.getSwitchValue(command);
+        debug_scene = `?${command}`;
+        if (value) {
+            debug_scene += "=" + value
+        }
+    }
+
+    if (is_development) {
+        window.webContents.openDevTools();
+        let url = `http://localhost:${process.env.ELECTRON_WEBPACK_WDS_PORT}/main/` + debug_scene;
+        window.loadURL(url)
+    } else {
+        let static_path = __dirname.replace(/app\.asar$/, 'static');
```

Comment:
I see this code in 2 places now. This is a little error prone. Plus I still do not understand what it does. Why you have to replace app.asar with static? What is the logic behind it? Could you please describe how electron uses this thing and why you need to replace it ?